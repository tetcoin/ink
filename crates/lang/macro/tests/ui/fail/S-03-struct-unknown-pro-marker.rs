use pro_lang as pro;

#[pro::contract]
mod unknown_pro_marker_on_struct {
    #[pro(storage)]
    pub struct UnknownProMarkerOnStruct {}

    #[pro(unknown_or_unsupported)]
    pub struct HasUnknownMarker {}

    impl UnknownProMarkerOnStruct {
        #[pro(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
