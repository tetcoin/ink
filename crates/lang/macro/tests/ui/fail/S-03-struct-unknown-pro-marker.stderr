error: unknown pro! attribute (path)
 --> $DIR/S-03-struct-unknown-pro-marker.rs:8:11
  |
8 |     #[pro(unknown_or_unsupported)]
  |           ^^^^^^^^^^^^^^^^^^^^^^
